Update LSP0 Pages (#461)
lukso-network/docs: docs/standards/universal-profile/lsp0-erc725account.md

## Introduction

> Externally Owned Accounts (EOAs) are the basic type of **account** in Ethereum controlled by a private key. If the private key is **compromised**, anyone can execute transactions from the account and access any assets it holds. As well, EOAs have no built-in mechanism for attaching any information or data, making it difficult to identify the person or entity using the account. They can only perform simple interactions using the **[CALL](https://www.evm.codes/#f1)** opcode and creating contracts using the **[CREATE](https://www.evm.codes/#f0)** opcode.

Reviewer: Applying Grammarly + maybe linking to Ethereum.org for good external references and SEO.

```suggestion
[Externally Owned Accounts (EOAs)](https://ethereum.org/en/developers/docs/accounts/#externally-owned-accounts-and-key-pairs) are the primary type of **account** in Ethereum, controlled by a private key. If the private key is **compromised**, anyone can execute transactions from the account and access any assets it holds. EOAs have no built-in mechanism for attaching any information or data, making identifying the person or entity using the account challenging. They can only perform simple interactions using the **[CALL](https://www.evm.codes/#f1)** opcode and create contracts using the **[CREATE](https://www.evm.codes/#f0)** opcode.
```

> These issues can be addressed by the **[ERC725](../lsp-background/erc725.md)** standard, which provides more operations to execute and a flexible way to attach data for the contract even after it's been deployed.

Reviewer: ```suggestion
These issues can be addressed by the **[ERC725](../lsp-background/erc725.md)** standard, which provides more operations to execute and a flexible way to attach data for the contract even after it has been deployed.
```

> However, for a smart contract-based account to be viable in the long term, it should have much more functionalities than the ability to execute and to attach data. The features that makes a smart contract an account are: the ability to **verify signed messages**, be **notified of incoming tokens**, followers, and other types of transactions, and be able to be **extended after deployment** to support functions and standards that will be adopted in the future. Additionally, it should have a **secure ownership management** system to ensure the protection of valuable assets.

Reviewer: Maybe we can break this down in bullet points?

--- 

The features that makes a smart contract an account are: 
- the ability to **verify signed messages**
- be **notified of incoming tokens**, followers, and other types of transactions,
- be able to be **extended after deployment** to support functions and standards that will be adopted in the future.

Additionally, it should have a **secure ownership management** system to ensure the protection of valuable assets.

- **[ERC1271](https://eips.ethereum.org/EIPS/eip-1271)** helps to **verify the validity** of a message and signature.

- **[LSP1-UniversalReceiver](../generic-standards/lsp1-universal-receiver.md)** enables **notifications about incoming or outgoing transactions** and adds custom handling and behavior based on these transactions.

- **[LSP14-Ownable2Step](../generic-standards/lsp14-ownable-2-step.md)** enables a **secure ownership management** system.

- **[LSP17-ContractExtension](../generic-standards/lsp17-contract-extension.md)** enables the contract to be **extended after deployment** to support new standard and functionalities.

### ERC725X - Generic Executor

:::tip

See the **[ERC725](../lsp-background//erc725.md)** standard for more information.

Check the [**execute functions**](../smart-contracts/lsp0-erc725-account.md#execute) provided by **ERC725X** that allows the contract to execute multiple operations.

Check the **javascript** guides to know [**How to Send native tokens**](../../guides/universal-profile/transfer-lyx.md) or [**How to Call other contract's function**](../../guides/universal-profile/interact-with-contracts.md) using the execute function.

:::

> The **ERC725X** standard enables the account to perform generic calls on other smart contracts, including transferring native tokens. External actions can be executed using the smart contract's generic `execute(...)` function and multi-calls can be done with the `execute(..)` **batch** function, but only the account owner can perform these operations.

Reviewer: We might need to discuss for this section, as we currently have the ERC725X and ERC725Y sections as duplicate in both LSP0 and the ERC725 page.

https://docs.lukso.tech/standards/lsp-background/erc725/#erc725x

This is not ideal as:
- the content is duplicate and we need to maintain in both places (doing the same work twice)
- in LSP0, there are illustrations, while in ERC725 there are no figures.
- we could end up having one page more up to date than the other.

To be discussed with @frozeman @Hugoo @YamenMerhi  . Let me know what you think.

Reviewer: agreed we keep it in both

### LSP1 - UniversalReceiver

:::tip

See the **[LSP1-UniversalReceiver](../generic-standards/lsp1-universal-receiver.md)** standard for more information.

Check the [**universalReceiver functions**](../smart-contracts/lsp0-erc725-account.md#universalreceiver) provided by **LSP1** that allows notifying about incoming/ outgoing transactions.

Check the **javascript** guides to know [**How to set a UniversalReceiverDelegate built by LUKSO**](../../guides/universal-receiver-delegate/set-default-implementation.md) or [**How to set your own UniversalReceiverDelegate**](../../guides/universal-receiver-delegate/accept-reject-assets.md).

:::

This standard enables the account to be notified of incoming transactions such as token transfer, vault transfer, information transfer, etc. Notifications are handy for situations where users want to customize how their account contract reacts to certain tokens by either rejecting them or operating a specific call on each token received.

The **[LSP0-ERC725Account](#)** implements the `universalReceiver(..)` function that:

> Emits an event with the `typeId` and `data` passed to it, as well as additional parameters such as the amount sent to the function, the caller of the function, and the return value of the delegate contracts.

Reviewer: We could add a link to the `UniversalReceiver` event in the contract ABI page.
The we might not even need to mention what is emitted in the event in this page, since it is mentioned in the contract ABI page for the event (or keep it simple to a minimum).

```suggestion
Emits an [`UniversalReceiver`](./link-to-lsp0-contract-abi-page-UR-event) event with the `typeId` and `data`, as well as additional parameters such as the amount sent to the function, the caller of the function, and the return value of the delegate contracts.
```

> The `typeId` is a **bytes32** that represents the type of action being notified about. For instance, if you want to notify an account about a specific type of token, you could hash the word **"TOKENXX"** which would result in a bytes32, and then use it as the typeId.

Reviewer: Suggesting adding tick for code to the other occurences for consistency.

```suggestion
The `typeId` is a **bytes32** value that represents the type of action being notified about. For instance, if you want to notify an account about a specific type of token, you could hash the word **"TOKENXX"** which would result in a `bytes32`, and then use it as a `typeId`.
```

> The `data` field would contain relevant information related to the typeId being notified about, in case of token it could be any encoded information such as its balance, time, royalties, etc.

Reviewer: ```suggestion
The `data` field can contain relevant information related to the `typeId` used when the `universalReceiver(...)` function was called. In the case of token transfers, it could be any encoded information such as the recipient balance, time, royalties, etc.
```

> The typeId provides a unique identifier for the type of notification, while the data field provides the specific information related to the notification. This allows for efficient and effective communication of information related to the token, without the need for extensive parsing or decoding of data.

Reviewer: I would suggest to merge the content from this in the two paragraphs above, for `typeId` and `data`

> Emitting the event is the mechanism through which an **LSP0-ERC725Account** can receive notifications. Websites can monitor and listen to the event and previous events to understand what the account has been notified about.

Reviewer: Suggesting to be more specific:

```suggestion
The `universalReceiver(...)` function and its `UniversalReceiver` event are the mechanisms through which an **LSP0-ERC725Account** can receive notifications. Websites can monitor and listen to the event and previous events to understand what the account has been notified about.
```
